index.js

```diff
@@ -308,23 +322,53 @@ Funnel.prototype.processFilters = function(inputPath) {
 
   this.outputToInputMappings = {}; // we allow users to rename files
 
-  if (this.files && !this.exclude && !this.include) {
-    entries = this._processPaths(this.files);
-    // clone to be compatible with walkSync
-    nextTree = FSTree.fromPaths(entries, { sortAndExpand: true });
+  // utilize change tracking from this.in[0]
+  var patches;
+  if (this._fsFacade) {
+    patches = this.in[0].changes();
+    // TODO: do we need this? if not, remove.
+    entries = this.in[0].entries;
+
+    patches.forEach(function(entry) {
+      var outputRelativePath = this.lookupDestinationPath(entry[2].relativePath);
+      this.outputToInputMappings[outputRelativePath] = entry[2].relativePath;
+      entry[1] = this.lookupDestinationPath(entry[1]);
+      entry[2].relativePath = outputRelativePath;
+    }, this);
+
+    if (this.destDir !== '/' || this.getDestinationPath) {
+      // add destination path to head of patches because it wont be in changes()
+      patches.unshift([
+        'mkdir',
+        this.lookupDestinationPath(''),
+        {
+          mode: 16877,
+          relativePath: path,
```

Comment:
Couple of questions here
1. path should be a string and should be same as this.lookupDestinationPath('')?
2. Shouldn't the entry also contain basePath?
3. What if the the relativePath has subdir1/subdir2 and both do not exist? In that case it should be mkdirp instead of mkdir??
Does it make sense?